src/main/java/br/com/metricminer2/scm/GitRepository.java

```diff
@@ -111,8 +119,12 @@ public SCMRepository info() {
 
 	}
 
-	protected Git openRepository() throws IOException {
-		return Git.open(new File(path));
+	protected Git openRepository() throws IOException, GitAPIException {
+		Git git = Git.open(new File(path));
+		if(this.masterBranchName == null) {
```

Comment:
The if test `masterBranchName == null` it's a kind of initalization, because `git.getRepository().getBranch()` could be expensive call since `openRepository()` is very often called.
Some repositories doesn't have "master" branch (eg. [https://github.com/mysql/mysql-server/branches]mysql), so `getBranch()` get the branch name witch HEAD points to.